src/dkg.ts

```diff
@@ -106,24 +112,57 @@ export class DkgClient {
-  //   const participants = await DkgCoordinatorAgent.getParticipants(
-  //     this.provider,
-  //     ritualId
-  //   );
-  //
-  //   const validatorMessages = participants.map((p) => {
-  //     const validatorAddress = EthereumAddress.fromString(p.provider);
-  //     const publicKey = FerveoPublicKey.fromBytes(fromHexString(p.???));
-  //     const validator = new Validator(validatorAddress, publicKey);
-  //     const transcript = Transcript.fromBytes(fromHexString(p.transcript));
-  //     return new ValidatorMessage(validator, transcript);
-  //   });
-  //   const aggregate = new AggregatedTranscript(validatorMessages);
-  //
-  //   return aggregate.verify(ritual.dkgSize, validatorMessages);
-  // }
+  public static async verifyRitual(
+    web3Provider: ethers.providers.Web3Provider,
+    ritualId: number
+  ): Promise<boolean> {
+    const ritual = await DkgCoordinatorAgent.getRitual(web3Provider, ritualId);
+    const participants = await DkgCoordinatorAgent.getParticipants(
+      web3Provider,
+      ritualId
+    );
+
+    // TODO: Does this check make sense here? Or do we delegate it to the Coordinator contract?
+    // for (const p of participants) {
+    //   // Not every participant has submitted a transcript
+    //   if (!p.aggregated) {
+    //     return false;
+    //   }
+    // }
+
+    const validatorMessages = participants.map((p) => {
+      const validatorAddress = EthereumAddress.fromString(p.provider);
+      // TODO: Replace with real keys
+      // const publicKey = FerveoPublicKey.fromBytes(fromHexString(p.???));
+      const publicKey = DkgClient.getParticipantPublicKey(p.provider);
+      const validator = new Validator(validatorAddress, publicKey);
+      return new ValidatorMessage(validator, p.transcript);
+    });
+    const aggregate = new AggregatedTranscript(validatorMessages);
+
+    return aggregate.verify(ritual.dkgSize, validatorMessages);
+  }
+
+  public static getParticipantPublicKey = (address: string) => {
+    // TODO: Without Validator public key in Coordinator, we cannot verify the
+    //    transcript. We need to add it to the Coordinator (nucypher-contracts #77).
+    const participantPublicKeys: Record<string, FerveoPublicKey> = {
+      '0x210eeAC07542F815ebB6FD6689637D8cA2689392': FerveoPublicKey.fromBytes(
+        fromHexString(
+          '6000000000000000ace9d7567b26dafc512b2303cfdaa872850c62b100078ddeaabf8408c7308b3a43dfeb88375c21ef63230fb4008ce7e908764463c6765e556f9b03009eb1757d179eaa26bf875332807cc070d62a385ed2e66e09f4f4766451da12779a09036e'
+        )
+      ),
+      '0xb15d5A4e2be34f4bE154A1b08a94Ab920FfD8A41': FerveoPublicKey.fromBytes(
+        fromHexString(
+          '60000000000000008b373fdb6b43e9dca028bd603c2bf90f0e008ec83ff217a8d7bc006b585570e6ab1ce761bad0e21c1aed1363286145f61134ed0ab53f4ebaa05036396c57f6e587f33d49667c1003cd03b71ad651b09dd4791bc631eaef93f1b313bbee7bd63a'
+        )
+      ),
+    };
```

Comment:
Can we do this in a test with mocks etc. instead of having hardcoded values in actual code and subsequently going stale?

Reply:
Until we fix https://github.com/nucypher/nucypher-contracts/pull/77 we can:
1) Expose this as a paremeter

Reply:
My vote would be the pause until https://github.com/nucypher/nucypher-contracts/pull/77 is dealt with.  This PR would then be a downstream change. Note that PR 77 would also have downstream effects on `nucypher`. 

I'd like to avoid hard-coding things that we know will change in actual code; tests are one thing, but in actual code it should be avoided if possible.

Reply:
I agree. Let's keep this PR reviewed open and reconsider merging after downstream changes are dealt with. I will document them where appropriate.